src/tabs/types.js

```diff
@@ -37,7 +37,8 @@ export type OnChangeHandlerT = ({activeKey: React.Key}) => mixed;
 
 export type TabsPropsT = {
   /** An array of Tab components. */
-  children: Array<React.Node>,
```

Comment:
## `React.Node` might be too loose

Using `React.Node` might be too loose, even though `React.Node` is being used as children for [some official types](https://github.com/facebook/flow/blob/0938da8d7293d0077fbe95c3a3e0eebadb57b012/lib/react.js#L265) and is supported as noted [here](https://stackoverflow.com/a/42887802) 

> [React.Node ](https://github.com/facebook/flow/blob/8fad62fe5016e201e0711a9a7916337487315bc4/website/en/docs/react/types.md#toc-react-node)can be null, a boolean, a number, a string, a React element, or an array of any of those types recursively.

## `React.Children` allows us  to enforce a specific type

Regarding using `React.Children`, this is the recommended method to [only allowing a specific element type as children](https://flow.org/en/docs/react/children/#toc-only-allowing-a-specific-element-type-as-children)

I saw this being used in [ baseweb's accordion component](https://github.com/uber-web/baseui/blob/0e4d7d28a71ac5c6b81d69bca9dcb081d38538fa/src/accordion/types.js#L14)

I am using the same logic here to enforce that the child is a [`React.Element` ](https://flow.org/en/docs/react/types/#toc-react-element) type which is defined as
> A React element is the type for the value of a JSX element

This will prevent null, boolean, number, and string from being rendered, which is desired. I can see a potential issue where someone may pass in a Tab override that conditionally renders `null` if the tab should be hidden in the case of a page that should only be visible to admins, but as a workaround they can render a `div` with style `display:'none'`

I intentionally made `React.Element<any>` so that when the Tab component class is overriden using overrides, it will still pass type checking.

Reply:
this type makes sense, but tbh I generally don't agree with using the tabs component with a single item (from a design perspective). i like that our types can enforce that restriction where consumers must opt-in to ignoring the flow error. don't feel too strongly about it though.. so if we can get ci to pass lgtm